tests/unit/hierarchy_test.py

```diff
@@ -0,0 +1,419 @@
+                ),
+            ),
+            learning_module_1=dict(
+                learning_module_class=EvidenceGraphLM,
+                learning_module_args=dict(
+                    max_match_distance=0.001,
+                    tolerances={
+                        "patch_1": {
+                            "hsv": np.array([0.1, 1, 1]),
+                            "principal_curvatures_log": np.ones(2),
+                        },
+                        "learning_module_0": {"object_id": 1},
+                    },
+                    feature_weights={"learning_module_0": {"object_id": 1}},
+                    max_graph_size=0.4,
+                    num_model_voxels_per_dim=200,
+                    max_nodes_per_graph=2000,
+                ),
+            ),
+        )
+
+        two_stacked_constrained_config = copy.deepcopy(base)
+        two_stacked_constrained_config.update(
+            experiment_args=SupervisedPretrainingExperimentArgs(
+                max_train_steps=30,
+                max_eval_steps=30,
+                max_total_steps=60,
+            ),
+            logging_config=PretrainLoggingConfig(
+                output_dir=self.compositional_save_path,
+                python_log_level="INFO",
+            ),
+            experiment_class=MontySupervisedObjectPretrainingExperiment,
+            monty_config=TwoLMStackedMontyConfig(
+                monty_args=MontyArgs(num_exploratory_steps=50),
+                learning_module_configs=two_stacked_constrained_lms_config,
+            ),
+            dataset_args=TwoLMStackedDistantMountHabitatDatasetArgs(
+                env_init_args=EnvInitArgsTwoLMDistantStackedMount(
+                    data_path=None,
+                ).__dict__,
+            ),
+        )
+
+        two_stacked_semisupervised_lms_config = copy.deepcopy(
+            two_stacked_constrained_config
+        )
+        two_stacked_semisupervised_lms_config.update(
+            experiment_args=SupervisedPretrainingExperimentArgs(
+                supervised_lm_ids=["learning_module_1"],
+                min_lms_match=2,
+                model_name_or_path=self.compositional_save_path + "/pretrained",
+            ),
+            monty_config=TwoLMStackedMontyConfig(
+                # set min_train_steps to 200 to send more observations to LM_1 after
+                # LM_0 has recognized the object.
+                monty_args=MontyArgs(min_train_steps=200, num_exploratory_steps=0),
+                learning_module_configs=two_stacked_constrained_lms_config,
+            ),
+        )
+
+        two_stacked_lms_eval_config = copy.deepcopy(two_stacked_constrained_config)
+        two_stacked_lms_eval_config.update(
+            experiment_args=ExperimentArgs(
+                do_train=False,
+                min_lms_match=1,
+                n_eval_epochs=2,
+                model_name_or_path=self.compositional_save_path + "/pretrained",
```

Comment:
This should use `os.path.join` or the newer `pathlib` equivalents to avoid any path separator differences between platforms. I've put off suggesting changing everything over to `pathlib` since there's a lot of the older style in the codebase, so I'm fine with using the `os.path` approach.
```suggestion
                model_name_or_path=os.path.join(self.compositional_save_path, "pretrained"),
```